packages/flame/lib/src/experimental/camera.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:meta/meta.dart';
+
+import '../components/component.dart';
+import 'max_viewport.dart';
+import 'viewfinder.dart';
+import 'viewport.dart';
+import 'world.dart';
+
+/// [Camera2] is a component through which a [World] is observed.
+///
+/// A camera consists of two main parts: a [Viewport] and a [Viewfinder]. It
+/// also a references a [World] component, and by "references" we mean that the
+/// world is not mounted to the camera, but the camera merely knows about the
+/// world, which may exist anywhere in the game tree.
+///
+/// A camera is a regular component that can be placed anywhere in the game
+/// tree. Most games will have at least one "main" camera for displaying the
+/// main game world. However, additional cameras may also be used for some
+/// special effects. These extra cameras may be placed either in parallel with
+/// the main camera, or even within the world itself. It is even possible to
+/// create a camera that looks at itself.
+///
+/// Since [Camera2] is a [Component], it is possible to attach other components
+/// to it. In particular, adding components directly to the camera is equivalent
+/// to adding them to the camera's parent. Components added to the viewport will
+/// be affected by the viewport's position, but not by its clip mask. Such
+/// components will be rendered on top of the viewport. Components added to the
+/// viewfinder will be rendered as if they were part of the world. That is, they
+/// will be affected both by the viewport and the viewfinder.
+class Camera2 extends Component {
```

Comment:
We had a discussion in the meeting now, maybe this can be named `AdvancedCamera` or something in the lines of that and live alongside the other simpler camera system? (We are going to try to get this merged in the beginning of this week)

Reply:
The new camera isn't that much more advanced than the old one; the main difference is that it is now a fully-fledged Component, so it can be added anywhere. So, perhaps call it `CameraComponent` then?

Reply:
Sounds good to me, I'll check with the rest.

Reply:
Some docs explaining the difference of the camera's would be good, maybe some docs about the other new concepts too, even though it is in the experimental folder.

Reply:
The others agreed to the `CameraComponent` name.

An export file called `experimental.dart` or such would be good to have too, and an explanation in the docs of what it means for a feature to be in there, so that people trying it don't have to do the internal imports.

My idea of the meaning of the experimental directory is to have things that we are not certain about the API etc, so that we can more easily make breaking changes regarding the features in there in the future. What do you think?